plugins/woocommerce/includes/admin/views/html-notice-contextual.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/**
+ * Admin View: Custom Notices
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+?>
+<div class="woocommerce-contextual-notice woocommerce-contextual-notice-<?php echo esc_attr( $severity ); ?>">
+  <?php echo wp_kses_post( $notice_html ); ?>
+  <div class="woocommerce-contextual-notice-cta" >
+  	<a class="components-button is-secondary"
+      href="<?php echo esc_url( $button_url ); ?>">
+      <?php echo esc_html( $button_text ) ?>
+    </a>
+  </div>
```

Comment:
I'd probably make the button parameters optional, and only conditionally render this part.

Reply:
I agree, hadn't got to that yet! Now all params except the body HTML are optional, and the button will only render if both params are provided. bb79e28cf6a7884c23613017c339d5e71d2a2a4c

We could also add a "dismissable" option. I'm hesitant to add it as I don't need it for my use case, and maybe these contextual notices should be reserved for things that shouldn't be dismissable. Can bring in design folks to refine how the new component should work & what it should allow.